meson: pylibmount missing Python.h include

Meson doesn't check for `Python.h` during the setup step when it is required for `pylibmount`. This leads to the following errors at compile time.

```
$ meson compile -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /var/home/jordan/Projects/util-linux/build
ninja: Entering directory `/var/home/jordan/Projects/util-linux/build'
[166/816] Compiling C object libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o
FAILED: libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o 
ccache cc -Ilibmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p -Ilibmount/python -I../libmount/python -Iinclude -I../include -Ilibmount -I../libmount -I/usr/include/python3.12 -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -fsigned-char -include /var/home/jordan/Projects/util-linux/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -fPIC -Wno-cast-function-type -Wno-error=redundant-decls -MD -MQ libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o -MF libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o.d -o libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o -c ../libmount/python/pylibmount.c
In file included from ../libmount/python/pylibmount.c:21:
../libmount/python/pylibmount.h:4:10: fatal error: Python.h: No such file or directory
    4 | #include <Python.h>
      |          ^~~~~~~~~~
compilation terminated.
[169/816] Compiling C object libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/fs.c.o
FAILED: libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/fs.c.o 
ccache cc -Ilibmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p -Ilibmount/python -I../libmount/python -Iinclude -I../include -Ilibmount -I../libmount -I/usr/include/python3.12 -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -fsigned-char -include /var/home/jordan/Projects/util-linux/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -fPIC -Wno-cast-function-type -Wno-error=redundant-decls -MD -MQ libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/fs.c.o -MF libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/fs.c.o.d -o libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/fs.c.o -c ../libmount/python/fs.c
In file included from ../libmount/python/fs.c:28:
../libmount/python/pylibmount.h:4:10: fatal error: Python.h: No such file or directory
    4 | #include <Python.h>
      |          ^~~~~~~~~~
compilation terminated.
[171/816] Compiling C object libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/tab.c.o
FAILED: libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/tab.c.o 
ccache cc -Ilibmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p -Ilibmount/python -I../libmount/python -Iinclude -I../include -Ilibmount -I../libmount -I/usr/include/python3.12 -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -fsigned-char -include /var/home/jordan/Projects/util-linux/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -fPIC -Wno-cast-function-type -Wno-error=redundant-decls -MD -MQ libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/tab.c.o -MF libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/tab.c.o.d -o libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/tab.c.o -c ../libmount/python/tab.c
In file included from ../libmount/python/tab.c:21:
../libmount/python/pylibmount.h:4:10: fatal error: Python.h: No such file or directory
    4 | #include <Python.h>
      |          ^~~~~~~~~~
compilation terminated.
```

Meson should properly check this header is available during the setup step.

--- a/libmount/python/meson.build
+++ b/libmount/python/meson.build
@@ -17,7 +17,7 @@ if build_python
     pylibmount_sources,
     include_directories : [dir_include],
     subdir : 'libmount',
-    dependencies : [mount_dep, python.dependency()],
+    dependencies : [mount_dep, python.dependency(embed: true)],
     c_args : [
       '-Wno-cast-function-type',
 

